### PR TITLE
Geodata by substation ids or line ids

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/GeoDataService.java [hunk 1]
```diff
@@ -38,29 +41,37 @@ public GeoDataService(@Value("${backing-services.geo-data.base-uri:http://geo-da
...
+    public String getLinesGraphics(UUID networkUuid, String variantId, List<String> linesIds) {
         var uriComponentsBuilder = UriComponentsBuilder.fromPath(DELIMITER + GEO_DATA_API_VERSION + "/lines")
             .queryParam(NETWORK_UUID, networkUuid);
 
         if (!StringUtils.isBlank(variantId)) {
             uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
         }
 
+        if (linesIds != null) {
+            uriComponentsBuilder = uriComponentsBuilder.queryParam(QUERY_PARAM_LINE_ID, linesIds);
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/study/server/service/GeoDataService.java:53]: You don't have to re-affect the builder as it is itsself that it returns

Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/GeoDataService.java]: Yes, done

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -293,19 +293,21 @@ public ResponseEntity<List<IdentifiableInfos>> getVoltageLevelBusbarSections(
...
 
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLinesGraphics(networkStoreService.getNetworkUuid(studyUuid), nodeUuid));
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLinesGraphics(networkStoreService.getNetworkUuid(studyUuid), nodeUuid, linesIds));
     }
 
     @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/geo-data/substations")
     @Operation(summary = "Get Network substations graphics")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The list of substations graphics")})
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The list of substations graphics with the given ids")})
     public ResponseEntity<String> getSubstationsGraphic(
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:305]: getSubstationsGraphic**s**

File: src/test/java/org/gridsuite/study/server/SingleLineDiagramTest.java [hunk 3]
```diff
@@ -364,11 +366,11 @@ public void testDiagramsAndGraphics() throws Exception {
 
         assertTrue(TestUtils.getRequestsDone(1, server).contains(String.format("/v1/lines?networkUuid=%s", NETWORK_UUID_STRING)));
 
-        mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/geo-data/lines/", studyNameUserIdUuid, modificationNodeUuid)).andExpectAll(
+        mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/geo-data/lines?lineId=LINEID", studyNameUserIdUuid, modificationNodeUuid)).andExpectAll(
                 status().isOk(),
                 content().contentType(MediaType.APPLICATION_JSON));
 
-        assertTrue(TestUtils.getRequestsDone(1, server).contains(String.format("/v1/lines?networkUuid=%s&variantId=%s", NETWORK_UUID_STRING, VARIANT_ID)));
+        assertTrue(TestUtils.getRequestsDone(1, server).contains(String.format("/v1/lines?networkUuid=%s&variantId=%s&lineId=LINEID", NETWORK_UUID_STRING, VARIANT_ID)));
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/study/server/SingleLineDiagramTest.java:373]: Add test with multiple values ?

File: src/main/java/org/gridsuite/study/server/service/GeoDataService.java [hunk 4]
```diff
@@ -38,29 +41,37 @@ public GeoDataService(@Value("${backing-services.geo-data.base-uri:http://geo-da
...
-    public String getLinesGraphics(UUID networkUuid, String variantId) {
+    public String getLinesGraphics(UUID networkUuid, String variantId, List<String> linesIds) {
         var uriComponentsBuilder = UriComponentsBuilder.fromPath(DELIMITER + GEO_DATA_API_VERSION + "/lines")
             .queryParam(NETWORK_UUID, networkUuid);
 
         if (!StringUtils.isBlank(variantId)) {
             uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
         }
 
+        if (linesIds != null) {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/service/GeoDataService.java:52]: We don't manage Country Query Parameter anywhere ?

**Review by sBouzols — COMMENTED** [review]
some comments

File: src/main/java/org/gridsuite/study/server/service/GeoDataService.java (hunk 4 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/service/GeoDataService.java:52]: No. It's implemented in geo-data-server but not used yet.

File: src/test/java/org/gridsuite/study/server/SingleLineDiagramTest.java (hunk 3 above)
Comment (etiennehomer) [reply src/test/java/org/gridsuite/study/server/SingleLineDiagramTest.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 5]
```diff
@@ -295,19 +295,21 @@ public ResponseEntity<List<IdentifiableInfos>> getVoltageLevelBusbarSections(
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The list of lines graphics")})
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:295]: ... with the given Ids, all otherwise"

Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

**Review by jonenst — APPROVED** [review]

**Pull request merged**